lib/modules/manager/npm/update/dependency/index.ts

```diff
@@ -117,13 +134,28 @@ export function updateDependency({
   logger.debug(`npm.updateDependency(): ${depType}.${depName} = ${newValue}`);
   try {
     const parsedContents: NpmPackage = JSON.parse(fileContent);
+    let overrideDepParents: string[] | undefined = undefined;
     // Save the old version
     let oldVersion: string | undefined;
     if (depType === 'packageManager') {
       oldVersion = parsedContents[depType];
       newValue = `${depName}@${newValue}`;
+    } else if (isOverrideObject(upgrade)) {
```

Comment:
```suggestion
    } else if (isOverrideObject(upgrade)) {
```

change the function declaration type to `UpdateDependencyConfig<NpmManagerData>`